v2/tools/generator/internal/astmodel/oneof_type.go

```diff
@@ -14,47 +14,100 @@ import (
 	"github.com/pkg/errors"
 )
 
-// OneOfType represents something that can be any
-// one of a number of selected types
+// OneOfType represents something that can be any one of a number of selected types
+// Early in processing, instances represent
+//  o  A base object (with discriminator property) that has a number of subtypes defined elsewhere
+//  o  A subtype (with discriminator value) that references a base object
+//  o  A complete object with subtypes nested within
+//
+// These three cases are unified into complete objects, and later converted into object definitions.
+//
 type OneOfType struct {
-	// invariants:
-	// - all types are unique (enforced by TypeSet)
-	// - length > 1
-	// - no nested OneOfs (aside from indirectly via TypeName)
-	types TypeSet
+	name                  string  // Name of the OneOf
```

Comment:
Renamed. It's used both to give the OneOf a better name (if we need to apply a name to an anonymous one) and to improve errors (so that we can search the original definitions for review).